packages/relay/src/lib/clients/mirrorNodeClient.ts

```
@@ -378,6 +378,16 @@ export class MirrorNodeClient {
             (data) => {
               // if the data is not valid, just return it to stick to the current behaviour
               if (data) {
+                if (
+                  typeof data === 'string' &&
+                  (data.trim().startsWith('<html') ||
+                    data.trim().startsWith('<!DOCTYPE html') ||
+                    data.trim().startsWith('<'))
```

Wouldn't be more resilient to just put `JSONBigInt.parse(data)` in a `try/catch` block?

This comparison is case sensitive, so for example `<HTML` would produce an error as well.

Note: this repeated `data.trim()` might cause performance problems. Of course, don't have hard data to back this, but this bit here is a hot spot, so having such comparison might cause problems.

---

Hey yeah I did try using try/catch but somehow I was clinging to the idea of throwing the parsing error which would discard the original error response from the Mirror Node. That’s why I opted to detect the issue before attempting to parse.

But looking back at the logic, the interceptor doesn’t need to throw a parsing error but just return the raw response so it can be handled later. Just updated it to follow that approach.

Thanks for raising this! It definitely makes more sense and is more robust than what I had before.